fair_mango/dataset/dataset.py

```diff
@@ -144,12 +175,96 @@ def plot_groups(
             plt.show()
 
     def get_data_for_all_groups(self) -> list[dict]:
-        """Retrieve data for all unique groups.
+        """Retrieve data corresponding to each demographic group present in
+        the sensitive features.
 
         Returns
         -------
         list[dict]
```

Comment:
improve the type of `dict`

Reply:
Done! the returned type is now `list[dict[str, np.ndarray | pd.DataFrame]]`